### PR TITLE
feat(`UPDATE`/`DELETE`): Enable path expressions for improved data manipulation

### LINKED ISSUE #159: @cap-js/sqlite: nested properties in where clause of delete crash
Hi CAP Team!

When deleting using nested properties:

```js
 const result = await DELETE.from(GroupMembers).where({'user.externalUserId': user.externalUserId })
```

I get the following error:
```sh
TypeError: Cannot read properties of undefined (reading 'map')
    at _entityTransitionsForTarget (/.../node_modules/@sap/cds/libx/_runtime/common/utils/resolveView.js:593:19)
```

In resolveView the  `from` input for function `_entityTransitionsForTarget` is:

```js
{
  join: "left",
  args: [
    {
      ref: [
        "user.db.GroupMembers",
      ],
      as: "GroupMembers",
    },
    {
      ref: [
        "user.db.Users",
      ],
      as: "user",
    },
  ],
  on: [
    {
      ref: [
        "user",
        "ID",
      ],
    },
    "=",
    {
      ref: [
        "GroupMembers",
        "user_ID",
      ],
    },
  ],
}
```
but in line 593 an attribute `ref` is expected, that is not there in the DELETE case.

```js
return from.ref.map((f, i) => { ... } // line 593
```

`from.ref` is undefined in this case.

| Your Project Name | N/A |
|:------------------|---------------------------------------|
| OData version     | v4                              |
| Node.js version   | v18.16.1                               |
| @sap/cds          | 7.0.2                                 |
| @sap/cds-compiler | 3.9.2                                |
| @sap/cds-dk       | 7.1.0                                 |
| @cap-js/sqlite | 1.0.1 |

Kind regards,
Sebastian

### PR DESCRIPTION
**Pull request by patricebender**
this change enables the usage of path expressions in `where` clauses of `UPDATE` and `DELETE` queries.

If there is a path expressions found in a `where` of an `UPDATE` or `DELETE`, the `where` clause is cut out of the query and is put into a `where` clause  in a dummy subselect. This subquery, selectes all primary keys. It is then transformed by `cqn4sql`. The result of the transformation is then put into the main queries `where` again, as `<list of primary key> in <subquery>`.

---

fixes #159

### CODE REVIEW

**Review by stewsk — APPROVED** [review]

**Review by johannes-vogel — APPROVED** [review]

File: db-service/test/cqn4sql/UPDATE.test.js
Comment (patricebender) [inline db-service/test/cqn4sql/UPDATE.test.js:89]: I did some manual, end-to-end testing with delete statements. It seems that an important peace is missing. Without the correlation to the outer query (`UPDATE` in this case), all entries will be matched and hence updated :(

Comment (BobdenOs) [reply db-service/test/cqn4sql/UPDATE.test.js:89]: My expectation would be something like this
```suggestion
    let expected = UPDATE.entity({ ref: ['bookshop.Books'] }).where(`
    ID IN (SELECT ID from bookshop.Books as Books
              left join bookshop.Authors as author on author.ID = Books.author_ID
              where author.name LIKE '%Bron%' or ( author.name LIKE '%King' and Books.title = 'The Dark Tower') and Books.stock >= 15
            )
  `)
```

Comment (patricebender) [reply db-service/test/cqn4sql/UPDATE.test.js:89]: cool idea, locally, I solved it by injecting the correlation into the `where` of the subquery

Comment (patricebender) [reply db-service/test/cqn4sql/UPDATE.test.js:89]: > Please also add a test for updating a projection. It's important that the SELECT is on the projection while the UPDATE's projection must be resolved to the table below.

what exactly does it mean? Does `cqn4sql` then learn what `resolveView` currently does, also w.r.t. renamings?

Comment (johannes-vogel) [reply db-service/test/cqn4sql/UPDATE.test.js:89]: No, that would simply cover the end-to-end scenario to work. Should be a test outside cqn4sql.

File: test/compliance/DELETE.test.js
Comment (patricebender) [inline test/compliance/DELETE.test.js:25]: added end2end test which works locally with the `resolveView` fix in the dev setup → skipped in this test suite for now as we need to wait for the next `cds` release.

Comment (johannes-vogel) [reply test/compliance/DELETE.test.js:25]: Can you please already prepare a PR to enable them? Just that we do not forget.

File: db-service/test/cqn4sql/DELETE.test.js
Comment (johannes-vogel) [inline db-service/test/cqn4sql/DELETE.test.js:161]: Why are there so many trailing commas?

Comment (patricebender) [reply db-service/test/cqn4sql/DELETE.test.js:161]: Because of this:
https://github.com/cap-js/cds-dbs/blob/7896289ec97f28bdc2c1d7700af1953608c96ef2/.prettierrc.js#L16

**Review by johannes-vogel — APPROVED** [review]

File: test/compliance/DELETE.test.js
Comment (patricebender) [reply test/compliance/DELETE.test.js:24]: #329

**Comment by patricebender** [comment]
please let me merge my PRs, I had some more cosmetic changes to do :(

**Comment by johannes-vogel** [comment]
> please let me merge my PRs, I had some more cosmetic changes to do :(

sry, I thought that's ready to merge.
Usually, I use the _draft PRs_ until they are ready to merge. Maybe we can do that as a general rule?

**Pull request merged**
